lib/ex_doc/doc_ast.ex

```diff
@@ -244,35 +298,43 @@ defmodule ExDoc.DocAST do
     end
   end
 
-  defp render_code(pre_attr, lang, lexer, lexer_opts, code, opts) do
-    highlight_tag = Keyword.get(opts, :highlight_tag, "span")
+  ## Traversal helpers
 
-    highlighted =
-      code
-      |> unescape_html()
-      |> IO.iodata_to_binary()
-      |> Makeup.highlight_inner_html(
-        lexer: lexer,
-        lexer_options: lexer_opts,
-        formatter_options: [highlight_tag: highlight_tag]
-      )
+  @doc """
+  Maps the tags in the AST, first mapping children tags, then the tag itself.
+  """
+  def map_tags({tag, attrs, inner, meta}, fun),
+    do: fun.({tag, attrs, Enum.map(inner, &map_tags(&1, fun)), meta})
```

Comment:
```suggestion
  def map_tags({tag, attrs, inner, meta}, fun) when is_list(inner),
    do: fun.({tag, attrs, Enum.map(inner, &map_tags(&1, fun)), meta})
    
  def map_tags({tag, attrs, inner, meta}, fun),
    do: fun.({tag, attrs, inner, meta})
```

When edoc is used to generate the doc ast, inner can be a string without being wrapped in a list.

Reply:
I basically wrote all of the traversal functions assuming that the contents is a list. :( Any chance we could normalize edoc in the long term? I will also improve the traversal functions here meanwhile.